src/PollingBlockTracker.ts

```diff
@@ -106,9 +120,33 @@ export class PollingBlockTracker
       return this._currentBlock;
     }
     // wait for a new latest block
-    const latestBlock: string = await new Promise((resolve) =>
-      this.once('latest', resolve),
-    );
+    const latestBlock: string = await new Promise((resolve, reject) => {
+      // eslint-disable-next-line prefer-const
+      let onLatestBlockUnavailable: InternalListener;
```

Comment:
this is first declared as uninitialized so that we can pass its reference to `onLatestBlockAvailable`